src/components/editor/services/monaco/formatters.ts

```diff
@@ -0,0 +1,73 @@
+import { Context, Effect, Layer } from "effect"
+import { createStreaming, type GlobalConfiguration } from "@dprint/formatter"
+import { type MonacoApi } from "../monaco"
+
+const globalConfig: GlobalConfiguration = {
+  indentWidth: 2,
+  lineWidth: 120
+}
+
+const typescriptPluginConfig = {
+  semiColons: "asi",
+  quoteStyle: "alwaysDouble",
+  trailingCommas: "never",
+  operatorPosition: "maintain",
+  "arrowFunction.useParentheses": "force"
+}
+
+const make = Effect.gen(function* () {
+  function setupCodeActions(monaco: MonacoApi) {
+    monaco.editor.addEditorAction({
+      id: "format",
+      label: "Format",
+      keybindings: [monaco.KeyMod.CtrlCmd | monaco.KeyCode.KeyS],
+      run: (editor) => {
+        const action = editor.getAction("editor.action.formatDocument")
+        if (action) {
+          action.run()
+        }
+      }
+    })
+  }
+
+  function setupTypeScriptFormatter(monaco: MonacoApi) {
+    return Effect.gen(function* () {
+      const formatter = yield* Effect.promise(() =>
+        createStreaming(
+          fetch("https://plugins.dprint.dev/typescript-0.90.5.wasm")
```

Comment:
@tim-smart - ideally I'd like to replace this with `/platform/Http`, but `createStreaming` expects a `ResponseLike` which looks like:

```ts
export interface ResponseLike {
    status: number;
    arrayBuffer(): Promise<BufferSource>;
    text(): Promise<string>;
    headers: {
        get(name: string): string | null;
    };
}
```

Reply:
This is fine :)